lisp/system.lisp

```diff
@@ -17,13 +17,19 @@ Mahogany is running under")
 		     (uiop:run-program (list "which" name) :output stream)))
     (UIOP/RUN-PROGRAM:SUBPROCESS-ERROR nil)))
 
+(defun open-program (candidates)
+  ;; TODO add log messages here for the case where no provided names work.
+  (loop :for name :in candidates
+		:do (alex:when-let ((program (find-program name)))
+              (uiop:launch-program program)
+              (return t)))
+  (values nil))
+
+(defvar *default-terminals*
+  `(,(uiop:getenv "TERMINAL")
+    "alacritty" "ghostty" "kitty" "xfce4-terminal"
+    "konsole" "gnome-terminal" "wezterm" "foot")
+  "A list of default terminal programs to use.")
+
 (defun open-terminal ()
-  (if-let* ((term (uiop:getenv "TERMINAL"))
-			(prog-path (find-program term)))
-      (uiop:launch-program prog-path)
```

Comment:
Can we keep this? This lets us change the environment variable while running and have the correct program be ran; putting it in the variable initialization means it's only read once during startup. 

Other than that this looks good.

Reply:
> Normally, a child process (e.g. terminal running in the WM) shouldn't be able to change the environment variables of the mother process (i.e. the WM).

You're right, but you can set the environment variables from inside the application via the repl. Here's what I currently have in my init file to pick the terminal app to use:
``` lisp
(setf (uiop:getenv "TERMINAL") "cosmic-term")
```

It's not the lispy way  to do it, but I would expect that running the above code would cause a change in behavior. For the sake of only having one "correct" way to do things, I would actually vote for not reading the environment variable at all. Let's just do that.